### ISSUE TITLE
TAStudio: Markers and Greenzone Broken in Dev Build

### ISSUE THREAD

**Issue by alyosha-tas** [TASVideos/BizHawk#2287]
I made this movie in dev build:

https://www.dropbox.com/s/lsybzi76v1ik6hk/Donkey%20Kong%20Land%202%20%28USA%2C%20Europe%29.bk2.tasproj?dl=0

Repro:
(the movie is for GBHawk, donkey kong land 2)
Load up the above movie in TAStudio (default settings), it should load to the last frame of the movie, 197954.
hit the '<<' button. 
Normally I would expect this to immediately go to the marker at138676, but instead it goes to 135136 and seeks from there.
Once it gets to the marker. click on a button on frame 138670, it will once again go to frame 135136 and start seeking forward again, even though all of the nearby frames are greenzoned since it just seeked by them.

I have no idea what the root cause is, but it doesn't seem to happen for short movies where everything is already greenzoned.

**Comment by adelikat** [comment]
The root cause, is that currently, markers are not guaranteed to be greenzoned.  That is the nature of using a ring buffer, the upside is lot of performance.

How important is it that a marker is guaranteed?  I'm still pondering how to achieve this functionality, unfortunately

**Comment by alyosha-tas** [comment]
In this particular case, the marker itself is not important, but the greenzone being broken is stopping me from doing any work.

**Comment by nattthebear** [comment]
Based on the informal, word of mouth requirements that I built `ZWinderStateManager` to, this requirement is complete nonsense.  You have absolutely no reason to be doing close navigation 60,000 frames off of the end of the movie, unless you're invalidating back to that point, in which case the buffer will start to refill with states from that new point.

Of course, I'm wrong and I'm sure this is a real requirement, but that just means that what I built is completely unsuitable to the task, and the "high priority" hack which nobody understands (including me) is probably not fixing anything either.

This state manager needs to be completely abandoned.

Closing in favor of https://github.com/TASVideos/BizHawk/issues/2288
